src/instance.rs

```diff
@@ -1114,14 +1129,16 @@ impl<'a, 'py> Borrowed<'a, 'py, PyAny> {
     /// derived from is valid for the lifetime `'a`.
     #[inline]
     pub(crate) unsafe fn from_ptr_unchecked(py: Python<'py>, ptr: *mut ffi::PyObject) -> Self {
-        Self(unsafe { NonNull::new_unchecked(ptr) }, PhantomData, py)
+        // SAFETY: caller has upheld the safety contract
+        unsafe { Self::from_non_null(py, NonNull::new_unchecked(ptr)) }
```

Comment:
We should add to the function safety docs, that `ptr` must be non-null (and a valid Python object)